### PR TITLE
openssl can fail, leaving an empty file and causing a reboot loop

### PR DESCRIPTION
**Pull request by yanokwa**
openssl can fail, leaving an empty file and causing a reboot loop since `/etc/dh/nginx.pem` cannot be found.

During an upgrade to Central v1.5.3, this is what the user saw when the server did not come up.
image

Logs showed it was an issue with `/etc/dh/nginx.pem`.
image

Not sure what happened, but it seems that for whatever reason, the openssl command failed and resulted in an empty file.

This PR checks for an empty file in all places where openssl is called. I confirmed the syntax is correct with shellcheck. I also wrap the mkdir command in case we ever use a white space in the name.

I confirmed this fix works on the server that had this issue.

### CODE REVIEW

**Comment by matthew-white** [comment]
@alxndrsn, if you have time, would you mind taking a look?

**Comment by matthew-white** [comment]
I'm going to target the `release` branch with the thought that this change would go out with v2022.3.

File: files/nginx/odk-setup.sh [hunk 1]
```diff
@@ -1,12 +1,12 @@
 DHPATH=/etc/dh/nginx.pem
-if [ ! -e "$DHPATH" ] && [ "$SSL_TYPE" != "upstream" ]
+if [[ (! -e "$DHPATH" || ! -s "$DHPATH" ) && "$SSL_TYPE" != "upstream" ]]
```
Comment (alxndrsn) [inline files/nginx/odk-setup.sh:2]: is `-e` made redundant by `-s`?

From the bash manpage:
```
       -s file
              True if file exists and has a size greater than zero.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+#### Why is this the best possible solution? Were any other approaches considered?
+
+#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
+
+#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/getodk/docs/issues/new) and include the link below.
+
+#### Before submitting this PR, please make sure you have:
+
+- [ ] verified that any code or assets from external sources are properly credited in comments.
```
Comment (alxndrsn) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: Changes seem unrelated to the PR title.

File: docs/news.html [hunk 3]
```diff
@@ -8,15 +8,15 @@
...
+        ODK Central v1.5.3
       </a>
     </div>
     <div class="news-item">
-      <span class="item-date">2022/07/12</span>
-      <a href="https://forum.getodk.org/t/odk-central-v1-5/37882/3" target="_blank">
-        ODK Central v1.5.2
+      <span class="item-date">2022/09/15</span>
+      <a href="https://forum.getodk.org/t/backups-to-google-drive-from-central-will-stop-working-after-jan-31st/38895" target="_blank">
+        ⚠️ Google backups sunsetted ⚠️
```
Comment (alxndrsn) [inline docs/news.html:19]: Changes seem unrelated to the PR title.

**Review by alxndrsn — COMMENTED** [review]
> I confirmed the syntax is correct with shellcheck. I also wrap the mkdir command in case we ever use a white space in the name.

I get two shellcheck violations:

```
In odk-setup.sh line 1:
DHPATH=/etc/dh/nginx.pem
^-- SC2148: Tips depend on target shell and yours is unknown. Add a shebang.


In odk-setup.sh line 12:
  mkdir -p $SELFSIGN_BASEPATH
           ^----------------^ SC2086: Double quote to prevent globbing and word splitting.
```

Is there a reason not to use bash's `-e` and `-o pipefail` options?  This might give more robust handling of unforeseen failures of individual commands in this file.

**Comment by yanokwa** [comment]
Sorry about the extra commits, I've cleaned that up and also fixed the redundant flag. The script should pass shellcheck.

As to `-e` and `-o pipefail`, maybe those can come as part of https://github.com/getodk/central/pull/328 so we can do that work at once on all scripts.

**Comment by yanokwa** [comment]
I went for the narrow change.

I don't know offhand how to fail immediately here. In what I've seen, the container will restart endlessly when something goes wrong.
